src/libraries/Microsoft.PowerFx.Core/Syntax/Visitors/DTypeVisitor.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Syntax.Visitors
+{
+    internal class DTypeVisitor : TexlFunctionalVisitor<DType, DefinedTypeSymbolTable>
```

Comment:
>DTypeVisitor [](http://example.com/codeflow?start=19&length=12)

Can we use existing visitor?

[https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Core/Public/Types/ITypeVisitor.cs](https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Core/Public/Types/ITypeVisitor.cs)